app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/AlertPlayer.java

```diff
@@ -19,8 +22,64 @@
 import com.eveningoutpost.dexdrip.Services.SnoozeOnNotificationDismissService;
 import com.eveningoutpost.dexdrip.SnoozeActivity;
 
+import java.io.IOException;
 import java.util.Date;
 
+// A helper class to create the mediaplayer on the UI thread. 
+// This is needed in order for the callbackst to work.
+class MediaPlayerCreaterHelper {
+    
+    private final static String TAG = AlertPlayer.class.getSimpleName();
+
+    Object lock1_ = new Object();
+    boolean mplayerCreated_ = false;
+    MediaPlayer mediaPlayer_ = null;
+    
+    MediaPlayer createMediaPlayer(Context ctx) {
+        if (isUiThread()) {
+            return new MediaPlayer();
+        }
+        
+        mplayerCreated_ = false;
+        mediaPlayer_ = null;
+        Handler mainHandler = new Handler(ctx.getMainLooper());
+
+        Runnable myRunnable = new Runnable() {
+            @Override 
+            public void run() {
+                synchronized(lock1_) {
+                    try {
+                        mediaPlayer_ = new MediaPlayer();
+                        Log.e(TAG, "media player created");
+                    } finally {
+                        mplayerCreated_ = true;
+                        lock1_.notifyAll();
+                    }
+                    
+                }
+            }
+        };
+        mainHandler.post(myRunnable);
+        
+        try {
+            synchronized(lock1_) {
+                while(mplayerCreated_ == false) {
```

Comment:
Can we be sure here, that we are not on the same Thread that the mainLooper is on, what could lead to a race condition where we block the runnable (mediaPlayer-producer) from being executed?